File: frontend/src/types/broadcast.ts
export interface AdvisoryPackage {
  name: string;
  version?: string;
  type?: string;
}

export type BroadcastTargetType = 'global' | 'teams' | 'advisory';
export type NotificationChannel = 'email' | 'slack' | 'mattermost' | 'teams';
export type BroadcastType = 'general' | 'advisory';

export interface BroadcastRequest {
  type: BroadcastType;
  target_type: BroadcastTargetType;
  target_teams?: string[];
  packages?: AdvisoryPackage[];
  subject: string;
  message: string;
  channels?: NotificationChannel[];
  dry_run?: boolean;
}

export interface BroadcastResult {
  recipient_count: number;
  project_count?: number;
  unique_user_count?: number;
}

export interface BroadcastHistoryItem {
  id: string;
  type: string;
  target_type: string;
  subject: string;
  created_at: string;
  created_by: string;
  recipient_count: number;
  project_count: number;
  unique_user_count?: number;
}

File: backend/app/models/broadcast.py
from app.models.base import CreatedAtModel
from app.models.types import MongoDocument


class Broadcast(MongoDocument, CreatedAtModel):
    type: str  # 'general' or 'advisory'
    target_type: str  # 'global', 'teams', 'advisory'
    subject: str
    message: str
    created_by: str  # user_id

    # Stats
    recipient_count: int = 0
    project_count: int = 0

    # Configuration
    packages: list[dict] | None = None
    channels: list[str] | None = None
    teams: list[str] | None = None
